tests/testthat/test-xgboost-develop.R

```diff
@@ -0,0 +1,76 @@
+context("Checking xgboost development")
```

Comment:
I got 6 warnings and 3 failures, all of which seemed related to SQLite or SQL Server:

Warnings ----------------------------------------------------------------------------------------------------------------------------------------------------
1. Lasso deploy regression pushes values to SQL Server (@test-deploy-pushes-to-sql-server.R#157) - ‘-’ not meaningful for factors

2. Lasso deploy regression pushes values to SQL Server (@test-deploy-pushes-to-sql-server.R#157) - ‘-’ not meaningful for factors

3. Lasso deploy regression pushes values to SQL Server (@test-deploy-pushes-to-sql-server.R#157) - ‘-’ not meaningful for factors

4. rf deploy regression pushes values to SQL Server (@test-deploy-pushes-to-sql-server.R#226) - ‘-’ not meaningful for factors

5. rf deploy regression pushes values to SQL Server (@test-deploy-pushes-to-sql-server.R#226) - ‘-’ not meaningful for factors

6. rf deploy regression pushes values to SQL Server (@test-deploy-pushes-to-sql-server.R#226) - ‘-’ not meaningful for factors

7. Failure: SQL Server - Returns correct selected data in data frame (@test-select-data.r#71) ---------------------------------------------------------------
selectData(MSSQLConnectionString = connection.string, query4) not equal to data.frame(...).
Component “SystolicBPNBR”: Modes: character, numeric
Component “SystolicBPNBR”: target is character, current is numeric


8. Failure: SQLite - inserts df successfully into table (@test-write-data.r#14) -----------------------------------------------------------------------------
writeData(df = df, SQLiteFileName = sqliteFile, tableName = "HCRWriteData") does not match "3 rows were inserted into the SQL Server HCRWriteData table.".
Actual value: "3 rows were inserted into the SQLite HCRWriteData table."


9. Failure: SQLite - insert df into table that didn't exist (@test-write-data.r#43) -------------------------------------------------------------------------
writeData(df = df, SQLiteFileName = sqliteFile, tableName = "HCRWriteData1") does not match "3 rows were inserted into the SQL Server HCRWriteData1 table.".
Actual value: "3 rows were inserted into the SQLite HCRWriteData1 table."